server/log.go

```diff
@@ -40,93 +30,93 @@ type Logger interface {
 // SetLogger sets the logger of the server
 func (s *Server) SetLogger(logger Logger, debugFlag, traceFlag bool) {
 	if debugFlag {
-		atomic.StoreInt32(&debug, 1)
+		atomic.StoreInt32(&s.logging.debug, 1)
 	} else {
-		atomic.StoreInt32(&debug, 0)
+		atomic.StoreInt32(&s.logging.debug, 0)
 	}
 	if traceFlag {
-		atomic.StoreInt32(&trace, 1)
+		atomic.StoreInt32(&s.logging.trace, 1)
 	} else {
-		atomic.StoreInt32(&trace, 0)
+		atomic.StoreInt32(&s.logging.trace, 0)
 	}
 
-	log.Lock()
-	log.logger = logger
-	log.Unlock()
+	s.logging.Lock()
+	s.logging.logger = logger
+	s.logging.Unlock()
 }
 
 // If the logger is a file based logger, close and re-open the file.
 // This allows for file rotation by 'mv'ing the file then signaling
 // the process to trigger this function.
 func (s *Server) ReOpenLogFile() {
 	// Check to make sure this is a file logger.
-	log.Lock()
-	ll := log.logger
-	log.Unlock()
+	s.logging.RLock()
+	ll := s.logging.logger
+	s.logging.RUnlock()
 
 	if ll == nil {
-		Noticef("File log re-open ignored, no logger")
+		s.Noticef("File log re-open ignored, no logger")
```

Comment:
Maybe not here, but as log was global and had sane defaults, it could be used safely even before server creation or when a server object failed to be created for some reason.